### PR TITLE
feat: update argocd versions used for testing

### PR DESCRIPTION
**Pull request by the-technat**
An attempt to get back to our promise of supporting the latest 2 versions of Argo CD.

Might require bumping libraries to new versions as well.

### CODE REVIEW

File: scripts/testacc_prepare_env.sh [hunk 1]
```diff
@@ -4,8 +4,8 @@ set -e
 
 export PATH=$PATH:.
 
-argocd_version=${ARGOCD_VERSION:-v2.8.13}
-k8s_version=${ARGOCD_KUBERNETES_VERSION:-v1.27.11}
```
[the remark this answers is not in the document]
Comment (the-technat) [reply scripts/testacc_prepare_env.sh:8]: You mean the K8s version? Or the topic in general?

**Comment by mkilchhofer** [comment]
Seems like Argo CD 2.12 is checking the repository:

https://github.com/argoproj-labs/terraform-provider-argocd/blob/582b426712a5b36cccdc628db411dc765ab58a0f/argocd/resource_argocd_repository_test.go#L196

```log
    resource_argocd_repository_test.go:45: Step 3/3 error: Error running apply: exit status 1
        
        Error: failed to update repository https://helm.nginx.com/stable
        
          with argocd_repository.helm,
          on terraform_plugin_test.tf line 19, in resource "argocd_repository" "helm":
          19: resource "argocd_repository" "helm" {
        
        rpc error: code = NotFound desc = repo 'https://helm.nginx.com/stable' not
        found
```

Maybe we pick argo-helm? :)

**Comment by the-technat** [comment]
To fix the failing test I tried updating the main argo-cd module using `go get github.com/argoproj/argo-cd/v2@v2.12.4`. But this in itself required other dependency updates.

So in the end I just ran:
```console
go get -u all
go mod edit -go=1.23
go mod tidy
go fix ./...
```

And now we are up-to-date with almost all dependencies. But that means we might be too new with some. For example `client-go` is now using the `v1.31.1` API schema of the K8s api.

**Comment by the-technat** [comment]
Will be combined / merged into #464

**Pull request closed without merging**